terraform/secrets.tf

```diff
@@ -0,0 +1,12 @@
+variable "subscription_id" {
```

Comment:
By having the secrets.tf part of the repository, it will lead to a situation where individuals need to add this file to their own .gitignore after downloading the repository, which is a recipe for someone committing their secrets to the repo.

Can we create a bash script that creates this file and/or handles grabbing these secrets, and is the entrypoint for running the terraform? For example, maybe having it copy /etc/secrets/gopherseattle.tf to the local repo, and then just add this file to the .gitignore from the beginning? This is not perfect, but it's a start. I am trying to avoid any mental overhead/steps required that are non-intuitive which may cause unintended behavior.

Reply:
You could also rename terraform/secrets.tf to terrform/secrets_template.tf or something, and then have a script that copies it to terraform/secrets.tf with empty values, and add terraform/secrets.tf to .gitignore from the beginning. Just throwing options at you.

Reply:
I put a .gitignore in the repository.  I am happy to change this if need by.  Which do you think is best?
I.E. I pushed up the sample, and then immediately added the secrets.tf to the .gitignore